src/fides/service/connection/merge_configs_util.py

```diff
@@ -232,3 +236,142 @@ def normalize_dataset(
 
     merged_dataset = upcoming_dataset_copy
     return merged_dataset
+
+
+def get_endpoint_resources(
+    db: Session, connection_config: ConnectionConfig
+) -> List[StagedResource]:
+    """
+    Get endpoint staged resources for a given connection config.
+    Only returns Endpoint type resources since those are what we need to preserve.
+
+    Returns:
+        List of endpoint staged resources
+    """
+    # Get all monitor configs for this connection
+    monitor_configs = MonitorConfig.filter(
+        db=db,
+        conditions=(MonitorConfig.connection_config_id == connection_config.id),
+    ).all()
+
+    if not monitor_configs:
+        return []
+
+    monitor_config_ids = [mc.key for mc in monitor_configs]
+
+    # Query staged resources for these monitor configs
+    # Only get Endpoint type resources since those represent SaaS endpoints/collections
+    endpoint_resources = StagedResource.filter(
+        db=db,
+        conditions=(
+            (StagedResource.monitor_config_id.in_(monitor_config_ids))
+            & (StagedResource.resource_type == "Endpoint")
+        ),
+    ).all()
```

Comment:
I think it's possible for a `StagedResource` to exist but not have it be "accepted" by the user. In this case, we wouldn't want to treat it as an endpoint to preserve. You'll have to verify this, or if this is the correct state once the endpoint/staged resource has been promoted.
```suggestion
    endpoint_resources = StagedResource.filter(
    db=db,
    conditions=(
        (StagedResource.monitor_config_id.in_(monitor_config_ids))
        & (StagedResource.resource_type == "Endpoint")
        & (StagedResource.diff_status == DiffStatus.MONITORED.value)
    ),
).all()
```